include/deal.II/grid/filtered_iterator.h

```diff
@@ -960,10 +960,8 @@ FilteredIterator<BaseIterator> &
 FilteredIterator<BaseIterator>::
 operator = (const FilteredIterator &fi)
 {
-  Assert ((fi.state() != IteratorState::valid) || (*predicate)(fi),
-          ExcInvalidElement(fi));
-  BaseIterator::operator = (fi);
-  return *this;
+  const BaseIterator &bi = fi;
+  return operator = (bi);
```

Comment:
The last interesting lines are
> 6400: include/deal.II/grid/tria_iterator.h: In instantiation of ‘dealii::TriaRawIterator<Accessor>::TriaRawIterator(const OtherAccessor&) [with OtherAccessor = dealii::FilteredIterator<dealii::TriaIterator<dealii::DoFCellAccessor<dealii::DoFHandler<2>, true> > >; Accessor = dealii::DoFCellAccessor<dealii::DoFHandler<2>, true>]’:

> 6400: include/deal.II/grid/tria_iterator.h:1110:31:   required from ‘dealii::TriaIterator<Accessor>::TriaIterator(const OtherAccessor&) [with OtherAccessor = dealii::FilteredIterator<dealii::TriaIterator<dealii::DoFCellAccessor<dealii::DoFHandler<2>, true> > >; Accessor = dealii::DoFCellAccessor<dealii::DoFHandler<2>, true>]’

> 6400: include/deal.II/grid/filtered_iterator.h:963:3:   required from ‘dealii::FilteredIterator<BaseIterator>& dealii::FilteredIterator<BaseIterator>::operator=(const dealii::FilteredIterator<BaseIterator>&) [with BaseIterator = dealii::TriaIterator<dealii::DoFCellAccessor<dealii::DoFHandler<2>, true> >]’

with the error in the PR description. Basically an implicit upcast is not performed in favor of trying to initialize `dealii::DoFCellAccessor<dealii::DoFHandler<2>, true>` with `dealii::FilteredIterator<dealii::TriaIterator<dealii::DoFCellAccessor<dealii::DoFHandler<2>, true> > >`